applications/GeoMechanicsApplication/tests/cpp_tests/test_convergence_criteria_factory.cpp

```diff
@@ -0,0 +1,59 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "testing/testing.h"
+#include "custom_utilities/convergence_criteria_factory.hpp"
+#include "spaces/ublas_space.h"
+#include "solving_strategies/convergencecriterias/mixed_generic_criteria.h"
+
+const std::string validParameters = R"(
+    {
+        "convergence_criterion":              "displacement_criterion",
+        "displacement_relative_tolerance":    1.0E-4,
+        "displacement_absolute_tolerance":    1.0E-9
+    }
+)";
+
+
+using namespace Kratos;
+using SparseSpaceType = UblasSpace<double, CompressedMatrix, Vector>;
+using LocalSpaceType = UblasSpace<double, Matrix, Vector>;
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(Create_ReturnsCorrectConvergenceCriteria_ForDisplacement, KratosGeoMechanicsFastSuite)
+{
+    const auto convergence_criteria = ConvergenceCriteriaFactory<SparseSpaceType, LocalSpaceType>::Create(Parameters{validParameters});
+    const auto displacementCriterion = dynamic_cast<DisplacementCriteria<SparseSpaceType, LocalSpaceType>*>(convergence_criteria.get());
```

Comment:
You may consider to add `const` here:
```suggestion
    const auto displacementCriterion = dynamic_cast<const DisplacementCriteria<SparseSpaceType, LocalSpaceType>*>(convergence_criteria.get());
```

Reply:
Agreed, done